src/Illuminate/Queue/CallQueuedHandler.php

```diff
@@ -121,6 +117,10 @@ protected function dispatchThroughMiddleware(Job $job, $command)
         return (new Pipeline($this->container))->send($command)
                 ->through(array_merge(method_exists($command, 'middleware') ? $command->middleware() : [], $command->middleware ?? []))
                 ->then(function ($command) use ($job) {
+                    if ($command instanceof ShouldBeUniqueUntilProcessing) {
```

Comment:
Here should it also check for ->released()? In case it was released
From within the handle method?